code/src/pages/MovieDetail.js

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect, useState } from 'react'
+import { useParams, Link, useHistory } from 'react-router-dom'
+import Company from '../pages/Companies'
+
+import BackButton from '../components/BackButton'
+const MovieDetail = () => {
+    const { id } = useParams()
+    const history = useHistory()
+    const [details, setDetails] = useState([])
+    const [statusCode, setStatusCode] = useState(200)
+    const [loading, setLoading] = useState(true)
+    const [prodComp, setProdComp] = useState([])
+
+    useEffect(() => {
+        fetch(`https://api.themoviedb.org/3/movie/${id}?api_key=2365aea36d60ef1f206bd1bdf23fd999&language=en-US`)
+        .then((res) => {
+            setStatusCode(res.status)
+            setLoading(false)
+            return res.json()  
+        }
+        )
+        .then((json) => {
+            setDetails(json)
+            setProdComp(json.production_companies)
+        })
+    },[])
+    useEffect(() => {
+        if (statusCode !== 200 || id.length > 19) {
+            history.push("/error")
+        }
+    }, [history, statusCode])
```

Comment:
We are curious whether there's a missing dependency in this array (id) because technically this useEffect depends on three variables: statusCode, id and history?🤓